nextstrain_profiles/nextstrain-open/builds.yaml

```diff
@@ -1,198 +1,107 @@
+auspice_json_prefix: ncov_open
+
 # Define custom rules for pre- or post-standard workflow processing of data.
 custom_rules:
   - workflow/snakemake_rules/export_for_nextstrain.smk
   - nextstrain_profiles/nextstrain/subsampling_ranges.smk
+  - nextstrain_profiles/nextstrain-open/auspice_configuration.smk
 
 # These parameters are only used by the `export_for_nextstrain` rule and shouldn't need to be modified.
 # To modify the s3 _source_ bucket, specify this directly in the `inputs` section of the config.
 # P.S. These are intentionally set as top-level keys as this allows command-line overrides.
-S3_DST_BUCKET: "nextstrain-data/ncov-intermediates"
+S3_DST_BUCKET: "nextstrain-data/files/ncov/open"
 S3_REF_BUCKET: "nextstrain-data/ncov-reference-datasets"
 S3_DST_COMPRESSION: "xz"
-S3_DST_ORIGINS: ["genbank"]
-
+S3_DST_ORIGINS: ["open"]
+
+# Build sizes are specific to this profile. For each specified build (defined below under `builds`)
+# we will create as many builds as there are keys in `build_sizes`, and the name of these builds
+# will be changed to have the build size appended (e.g. `global` -> `global-xsmall`).
+# (Note: the build size `standard` is not appended, i.e. the original build name remains unchanged.)
+# Each build will have it's own subsampling scheme, which is a copy of the one specified by
+# the build but with the build size appended (e.g. build `global` specifies `nextstrain_region_global`,
+# build `global-xsmall will specify `nextstrain_region_global_xsmall`).
+# The copy of this scheme will be the same as defined in this file with one difference:
+# any `max_sequences` parameters less than 1.0 will be multiplied by the value supplied below. 
 build_sizes:
   standard: 4000
   small: 2000
   xsmall: 1000
```

Comment:
I'm not convinced that we need three different sizes per build. I understand that this was a feature of the original implementation, but I don't know the specific users who benefit from these different builds that wouldn't just as easily benefit from a single standard build.

I'm also worried that automating build definitions like we do here encourages us to hardcode configuration-level details in Snakemake source code (as in the `auspice_configuration.smk` above or in the [`subsampling_ranges.smk`](https://github.com/nextstrain/ncov/blob/master/nextstrain_profiles/nextstrain/subsampling_ranges.smk)). Just like any other software we develop, hardcoding configuration details in the source code makes those details difficult to discover and debug. In both of the `.smk` files I mention above, the corresponding functionality could be implemented as simpler YAML configuration if `augur` provided a couple of additional flags for `export` and `filter`, respectively.

As programmers, I think we see a hardcoded list of 50 states (or 7 regions in this case) and think we should automate the solution. In practice, a verbose configuration file is just as easy to generate once and then maintain as a static file. The frequency these files need to be updated does not require automated tooling to produce the updated files. A good example of this is [the SPHERES US builds file](https://github.com/CDCgov/spheres-augur-build/blob/master/spheres_profile/builds.yaml) that I generated once and that hasn't needed modification since.

My general thinking here is that workflows like this are complicated enough to follow and debug even in their simplest form. When we add automated configuration to the workflow, we make it that much harder to understand in the future for the short-term benefit of not manually editing verbose configuration files.

Reply:
Yup! This is why I added 9 lines of comments as I knew I wouldn't remember what was going on next time I looked through the config 😉

Reply:
this is why I explored some way to template builds and a combinatorial way:

https://github.com/nextstrain/ncov-simple/blob/master/workflow/snakemake_rules/templated_build.smk

this isn't trying to make an argument for three different sizes, but generating the regions builds only takes 40 lines including full subsampling specs:
https://github.com/nextstrain/ncov-simple/blob/master/profiles/genbank/builds.yaml#L16-L56